### PR TITLE
2269 handle exploitation events

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do?

Adds a subscriber that updates the nodes on exploit events

#2269

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running unit tests
    > Tested by running a simple simulation and verifying the contents of the database
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/monkey_island/cc/agent_event_handlers/update_nodes_on_exploitation.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from monkey_island.cc.repository import NetworkModelUpdateFacade
+
+
+class update_nodes_on_exploitation:
+    def __init__(self, network_model_update_facade: NetworkModelUpdateFacade):
+        self._network_model_update_facade = network_model_update_facade
+
+    def __call__(self, event: ExploitationEvent):
+        if event.success:
+            self._network_model_update_facade.upsert_communication_from_event(
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/agent_event_handlers/update_nodes_on_exploitation.py:12]: We should also update the machine with relevant services. It makes no sense that we used log4j to exploit tomcat on 8080, but still don't know what kind of service is running on 8080

**Review by VakarisZ — APPROVED** [review]
Looks good. Only concern that pops to mind is that

File: monkey/tests/unit_tests/monkey_island/cc/agent_event_handlers/test_update_nodes_on_exploitation.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    handler = update_nodes_on_exploitation(network_model_update_facade)
+
+    handler(SUCCESSFUL_EXPLOITATION_EVENT)
+
+    network_model_update_facade.upsert_communication_from_event.assert_called_with(  # type: ignore[attr-defined]  # noqa E501
+        SUCCESSFUL_EXPLOITATION_EVENT, CommunicationType.EXPLOITED
+    )
+
+
+def test_no_update_on_failed_expliot(network_model_update_facade: NetworkModelUpdateFacade):
```
Comment (shreyamalviya) [inline monkey/tests/unit_tests/monkey_island/cc/agent_event_handlers/test_update_nodes_on_exploitation.py:44]:
```suggestion
def test_no_update_on_failed_exploit(network_model_update_facade: NetworkModelUpdateFacade):
```

**Review by shreyamalviya — APPROVED** [review]

File: monkey/monkey_island/cc/agent_event_handlers/update_nodes_on_exploitation.py (hunk 1 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/agent_event_handlers/update_nodes_on_exploitation.py:12]: Agreed, but not now. Somewhere down the road.

**Pull request merged**